### PR TITLE
[wait for #1783] [lr] Support step wise decay scheduler 

### PR DESCRIPTION
**Pull request by kparichay**
Support step wise decay learning rate scheduler, where the iterations
and learning rate are taken as parameter.
The corresponding unittests are also added.

See Also #1776

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: nntrainer/optimizers/lr_scheduler_step.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+ *
+ * @file   lr_scheduler_exponential.cpp
+ * @date   09 December 2021
+ * @brief  This is Exponential Learning Rate Scheduler class
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug    No known bugs except for NYI items
+ *
+ * @details given the iteration range [it_0, it_1, ... it_n], learning rates
+ * [lr_0, lr_1 ... lr_n_1], and iteration iter, find index i such that
```
Comment (lhs8928) [inline nntrainer/optimizers/lr_scheduler_step.cpp:13]: lr_n+1?

File: nntrainer/optimizers/lr_scheduler_step.cpp [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+ * it_i_1 < iter <= it_i and return lr_i_1.
```
Comment (lhs8928) [inline nntrainer/optimizers/lr_scheduler_step.cpp:14]: it_i-1, lr_i-1?

**Review by lhs8928 — APPROVED** [review]
LGTM

**Comment by jijoongmoon** [comment]
This pr is included in #1871
